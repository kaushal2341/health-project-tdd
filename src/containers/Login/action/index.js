import {login} from './constants';
export const authenticate = (data) => {
     return {
     payload:data,
     type:login.AUTH
     }
};