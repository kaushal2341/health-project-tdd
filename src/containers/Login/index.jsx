import React from 'react';
import { Form, Message } from 'semantic-ui-react';
import { TextInput, PasswordInput, ButtonInput } from '../../common-ui';
import { postRaw } from '../../axios';
export default class Login extends React.PureComponent {
  state = {
    userCredential: '',
    password: '',
    errorMsg: []
  }
  setTheState = (name, value) => {
    this.setState((prevState) => ({
      [name]: value
    }))
  }

  onChangeHandler = e => {
    const { name, value } = e.target;
    this.setTheState(name, value);
  }

  checkFormSubmitIsValid = () => {
    let errorMsg = [...this.state.errorMsg];
    let formStatus = false;
    Object.keys(this.state).map((s, i) => {
      if (!Object.values(this.state)[i] && s !== 'errorMsg') {
        errorMsg[i] = ` ${s.toUpperCase()} should not be empty`;
        formStatus = false;
      }
      else {
        const patt = /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/;
        const value = Object.values(this.state)[i];
        if (s === 'password' && value.length <= 5) {
          errorMsg[i] = `${s.toUpperCase()} length should be equal or greater than 6`;
          formStatus = false;
        }
        else if (s === 'username' && value.includes('@') && !value.match(patt)) {
          errorMsg[i] = `Not a valid ${s.toUpperCase()}/EMAIL `;
          formStatus = false;
        }
        else {
          errorMsg[i] = '';
          formStatus = true;
        }
      }
    });

    this.setTheState('errorMsg', errorMsg);
    return formStatus;
  }

  apiCall = () => {
    const data = {
      userCredential: this.state.userCredential,
      password: this.state.password
    }
    
    if(this.props.getTokenAfterLogin){
       console.log(data)
       this.props.getTokenAfterLogin(data)
    }   
    // postRaw('/api/login', data).then(response =>{
    //   this.props.submitHandler(response.data);  
    //   console.log('helllo');
    // }).catch(e => {
    //   console.log('error');
    // })  
  }

  submitFormHandler = e => {
    e.preventDefault();
    let isFormValid = this.checkFormSubmitIsValid();
    if (isFormValid)
         this.apiCall();
  }

  render() {
    const depts = [
    { key: '1', text: "Admin", value: "1" },
    { key: '2', text: "Pharmacy", value: "2" },
    { key: '3', text: "Finance", value: "3" }
    ]
    const errorMsg = this.state.errorMsg;
    
    return (
      <>
        <Form>
          <TextInput placeholder="Username/Email" name="userCredential" type="text" onChangeHandler={this.onChangeHandler} value={this.state.userCredential}></TextInput>
          <PasswordInput placeholder="Password" name="password" onChangeHandler={this.onChangeHandler} value={this.state.password}></PasswordInput>
          <ButtonInput id="f-submitter" type='submit' buttonName="Submit" onClickHandler={this.submitFormHandler} ></ButtonInput>
          <Message negative>
            {errorMsg.map((msg, i) => {
              return msg ? <p key={i} >{msg}</p> : ""
            })}
          </Message>
        </Form>
      </>
    )
  }
}

