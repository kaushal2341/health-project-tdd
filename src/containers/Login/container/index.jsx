import React from 'react';
import Login from '../../';
import {connect} from 'react-redux';
import {authenticate} from '../action'
class LoginContainer extends React.PureComponent {
 render(){
     return(
         <Login></Login>
      )
 }
}
const mapStateToProps = state => ({
token:state.token
})
const mapDispatchToProps = dispatch => ({
  login:(data) => authenticate(data)
})

export default connect(mapStateToProps,mapDispatchToProps)(LoginContainer)