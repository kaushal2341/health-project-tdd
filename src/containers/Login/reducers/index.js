import {login} from '../action/constants';        
let state={};

export const loginState =(state,action) =>{
    switch(action.type) {
        case login.AUTH:console.log(state)
                                return ({...state,
                                         token:action.payload
                                         })
        default:return state;
    }
 

};