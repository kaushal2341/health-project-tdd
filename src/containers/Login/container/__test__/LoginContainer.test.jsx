import React from 'react';
import LoginContainer from '..';
import Login from '../../';
import configureMockStore from 'redux-mock-store';
expect.addSnapshotSerializer(enzymeSerializer)
describe('Login Container', () => {
            let wrapper, mockStore, store, mockLoginfn;
            beforeEach(() => {
                    ;
                    mockLoginfn = jest.fn();
                    let initialState = {
                        token: ''
                    }
                    mockStore = configureMockStore();
                    store = mockStore(initialState);
                    wrapper = shallow(<LoginContainer store={store}/>)
                    });

                it('should be defined', () => {
                    expect(wrapper).toBeDefined();
                });

                it('should have one Login Component', () => {
                    console.log(wrapper.props())
                    expect(wrapper.props().children.props.login).toBeDefined();
                });

                it('should have state mapped into props connected to store',() => {
                    expect(wrapper.props().children.props.token).toBeDefined();
                })

                describe('Snap Shot Testing', () => {
                    it('should create or match previous snapshot of the component', () => {
                      expect(wrapper).toMatchSnapshot();
                    });
                  });

            })