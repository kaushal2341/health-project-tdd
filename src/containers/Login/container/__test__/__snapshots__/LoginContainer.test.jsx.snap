// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container Snap Shot Testing should create or match previous snapshot of the component 1`] = `
<ContextProvider
  value={null}
>
  <LoginContainer
    login={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    token=""
  />
</ContextProvider>
`;
